cmd/rpc.go

```diff
@@ -69,6 +87,36 @@ func InitClient(cmd *cobra.Command, _ []string) error {
 	return nil
 }
 
+func getStorePath(cmd *cobra.Command) (string, error) {
+	// if node store flag is set, use it
+	if cmd.Flag(nodeStoreFlag).Changed {
+		return cmd.Flag(nodeStoreFlag).Value.String(), nil
+	}
+
+	errMsg := "cant get the access to the auth token: token/node-store flag was not specified"
+
+	// try to detect a running node
+	path, err := nodebuilder.DiscoverOpened()
+	if err != nil {
+		return "", fmt.Errorf("%s: %w", errMsg, err)
+	}
+
+	if path != "" {
+		return path, nil
+	}
+
+	return "", errors.New(errMsg)
+}
+
+func determineProtocol(address string) (protocol string, trimmedAddress string) {
```

Comment:
@Wondertan is this needed or http as default is fine? lmk.

Reply:
You could add a little method to the RPC configuration struct that gives you the full address. There, you would encapsulate logic that builds an address. 

Also, to answer your question, the logic can trim any protocol prefix(by looking at '://') and use it as is. You don't need to detect if that HTTP or https.